### ISSUE TITLE
"npm run package" returns "npm ERR!" with Object.values & Object.entries

### ISSUE THREAD

**Issue by Avolition** [irvinlim/es2017-lambda-boilerplate#20]
Hi Irvin. Glad I came across your package as callback hell was really slowing down my current lambda project.

I receive the following error when attempting "npm run package" on a script which utilizes Object.values || Object.entries. The entire script + async / await appear to work fine when uploaded to AWS without either Object methods.
Terminal output:

>npm run package

> es2017-lambda-boilerplate@0.3.1 package /Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI
> npm run build && npm run package:pack

> es2017-lambda-boilerplate@0.3.1 build /Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI
> npm run build:init && npm run build:js && npm run build:install

> es2017-lambda-boilerplate@0.3.1 build:init /Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI
> rm -rf dist && mkdir dist

> es2017-lambda-boilerplate@0.3.1 build:js /Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI
> cd src && babel . -d ../dist

>index.js -> ../dist/index.js
TypeError: main.js: Property body[1] of BlockStatement expected node to be of a type ["Statement"] but instead got "FunctionExpression"
    at validate (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/definitions/in>dex.js:109:13)
    at validator (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/definitions/in>dex.js:57:7)
    at Object.validate (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/definitions/in>dex.js:190:10)
    at validate (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/index.>js:505:9)
    at Object.builder (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-types/lib/index.>js:466:7)
    at NodePath.replaceExpressionWithStatements (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-traverse/lib/path/replac>ement.js:206:54)
    at NodePath.insertBefore (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-traverse/lib/path/modifi>cation.js:52:10)
    at NodePath.unshiftContainer (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-traverse/lib/path/modifi>cation.js:239:15)
    at File.addHelper (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-core/lib/transformation/>file/index.js:388:17)
    at PluginPass.addHelper (/Users/arg/CloudDrive/Projects/Personal/CryptoLAMBDA/Binance_CLI/node_modules/babel-core/lib/transformation/>plugin-pass.js:46:42)
>npm ERR! code ELIFECYCLE
>npm ERR! errno 1
>npm ERR! es2017-lambda-boilerplate@0.3.1 build:js: 'cd src && babel . -d ../dist'
>npm ERR! Exit status 1
>npm ERR!
>npm ERR! Failed at the es2017-lambda-boilerplate@0.3.1 build:js script.
>npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

>npm ERR! A complete log of this run can be found in:
>npm ERR!     /Users/arg/.npm/_logs/2018-01-27T07_33_28_532Z-debug.log
>npm ERR! code ELIFECYCLE
>npm ERR! errno 1
>npm ERR! es2017-lambda-boilerplate@0.3.1 build: 'npm run build:init && npm run build:js && npm run build:install'
>npm ERR! Exit status 1
>npm ERR!
>npm ERR! Failed at the es2017-lambda-boilerplate@0.3.1 build script.
>npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

>npm ERR! A complete log of this run can be found in:
>npm ERR!     /Users/arg/.npm/_logs/2018-01-27T07_33_28_548Z-debug.log
>npm ERR! code ELIFECYCLE
>npm ERR! errno 1
>npm ERR! es2017-lambda-boilerplate@0.3.1 package: 'npm run build && npm run package:pack'
>npm ERR! Exit status 1
>npm ERR!
>npm ERR! Failed at the es2017-lambda-boilerplate@0.3.1 package script.
>npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

>npm ERR! A complete log of this run can be found in:
>npm ERR!     /Users/arg/.npm/_logs/2018-01-27T07_33_28_565Z-debug.log

[I've added your email to my private repo if you need to checkout the source file](https://bitbucket.org/avolition/cryptolambda/src/31eece9d23c807c06135b50884c89b3f5d4db110/Binance_CLI/src/main.js?fileviewer=file-view-default).

**Comment by irvinlim** [comment]
Hi @Avolition, let me take a look at your repo and get back to you. You're right that `Object.values/entries` is supposed to be transpiled down to ES6-compatible shims.

**Comment by irvinlim** [comment]
Could you add my other email address instead? Check the raw git commits in this repository for my main email address (Gmail). I can't access the repository even though it's an email alias, thanks!

**Comment by Avolition** [comment]
Yup done. FYI I don't actually need these object methods for my current project; was just testing them out and figured I'd let you know, so feel free to take your time figuring out what the problem is.

**Comment by irvinlim** [comment]
Okay, I managed to reproduce it, but haven't found a good solution yet. Just some updates for you (and myself or others) to follow:

Turns out that Babel transpilation of modules was interfering with [babel-plugin-transform-es2017-object-entries](https://github.com/christensson/babel-plugin-transform-es2017-object-entries), though I'm not exactly sure why. If I modify `.babelrc` so that [module transformation](https://www.npmjs.com/package/babel-preset-env#modules) is disabled in babel-preset-env:

```json
{

    "presets": [
        [
            "env",
            {
                "targets": { "node": "6.10.0" },
                "modules": false
            }
        ]
    ],
}
```

However, if I disable module transformation (that is, disable transforming `import`/`export` to `module.exports` and `require`), then a bulk of the existing boilerplate would fail since I used `import` quite extensively. Node.js 6.10 doesn't support ES6 modules without the [`--experimental-modules` flag](https://nodejs.org/api/esm.html), and so AWS Lambda wouldn't support it as well.

At this point, I can either try and find out why babel-plugin-transform-es2017-object-entries isn't working with ES6 module transformation, or switch all instances of `import`/`export` to CommonJS-style modules.

**Comment by irvinlim** [comment]
Turns out, it's because I used `export default` without including [babel-plugin-transform-es2015-modules-commonjs](https://babeljs.io/docs/plugins/transform-es2015-modules-commonjs/). Only  babel-plugin-transform-es2017-object-entries ran into an error when used in conjunction with ES6 Modules without the appropriate transform.

**Comment by irvinlim** [comment]
Fixed in 0.4.0.

**Comment by Avolition** [comment]
Great job! I'll test out the latest v with my next lambda fn and let you know if I encounter any further issues. Cheers!
